src/rebar_prv_shell.erl

```diff
@@ -331,8 +331,8 @@ reread_config(State) ->
     case find_config(State) of
         no_config ->
             ok;
-        ConfigList ->
-            _ = rebar_utils:reread_config(ConfigList),
+        Config ->
+            _ = rebar_utils:reread_config([Config]),
```

Comment:
Hm, because there can be recursive lists, I guess we could end with
`[ConfigA, ConfigVar1, [ConfigBVar1, [ConfigCVar1,...], ConfiguBVar2], ...]` and so on as they are nested. Understood.

So the interesting bit here is that there's two pathways possible in this thing:

1. people are loading the configs from `rebar_file_utils:consult_config` into `rebar_utils:reread_config`, which means their current use case will break without changing code
2. people are manually loading lists, which means the new format change will break for them

I guess the proper fix in this case would be to change `reread_config` so it does not require an API change when taking the value passed off `rebar_file_utils:consult_config/2`. We can then keep the flatmap, but would need to wrap it into an additional list at the producing site (instead of the callsite), so the callers do not break (and the type signature remains intact). 

This fixes 1). Because we also do not change the other function, we ensure that we do this without messing up the other usages of either function in the wild with 2).